src/Bridge/Doctrine/MongoDB/CollectionDataProvider.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDB;
+
+use ApiPlatform\Core\Api\CollectionDataProviderInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\ODM\MongoDB\DocumentRepository;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryResultExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\MongoDB\Extension\QueryCollectionExtensionInterface;
+
+/**
+ * Collection data provider for the Doctrine MongoDB ODM.
+ */
+class CollectionDataProvider implements CollectionDataProviderInterface
+{
+    private $managerRegistry;
+    private $collectionExtensions;
+    private $decorated;
+
+    /**
+     * @param ManagerRegistry                      $managerRegistry
+     * @param QueryCollectionExtensionInterface[]  $collectionExtensions
+     * @param CollectionDataProviderInterface|null $decorated
```

Comment:
My personal preference is to inject the decorated object as the first argument, so we can easily see that it's a decorator.